move_group/src/default_capabilities/move_action_capability.cpp

```diff
@@ -94,6 +93,7 @@ void move_group::MoveGroupMoveAction::executeMoveCallback_PlanAndExecute(const m
 
   if (planning_scene::PlanningScene::isEmpty(goal->planning_options.planning_scene_diff))
   {
+    context_->planning_scene_monitor_->syncSceneUpdates();
```

Comment:
On Thu, Jul 21, 2016 at 05:47:12PM -0700, Robert Haschke wrote:
> When distributing `syncSceneUpdates()` throughout the code, I mainly looked for instances of `LockedPlanningSceneRO`, which indicated existing use of a (hopefully up-to-date) planning scene.

This is not enough! The PSM has a (unsafe) `getPlanningScene` method that is used in a number of places.
I would propose to explicitly sync only in very few places in order to reduce further slow-down.

- right in the beginning of all interfaces that are used to Plan or Plan and Execute and allow (or dictate) to specify an empty start state.
We need this to ensure the interfaces really use the *current* state.

- in the beginning of interfaces to execute a trajectory, if we want to validate the trajectory begins at the current state.

Do we need to know the *current* robot-state anywhere else?